ESP32/main/main.ino

```diff
@@ -1,565 +1,190 @@
 #endif
 
-/*
-   Register devices
-*/
-AccelStepper stepper_A;
-AccelStepper stepper_X;
-AccelStepper stepper_Y;
-AccelStepper stepper_Z;
+#include "soc/soc.h"
+#include "soc/rtc_cntl_reg.h"
 
-#ifdef IS_SLM
-#include "parameters_slm.h"
-#endif
+PINDEF *pins;
 
-/* --------------------------------------------
-   Setup
-  --------------------------------------------
-*/
+static const char *TAG = "Main.ino";
 void setup()
 {
-  /*
-     SETTING UP DEVICES
-  */
-
-
-
-  WRITE_PERI_REG(RTC_CNTL_BROWN_OUT_REG, 0); //disable brownout detector
-
-  // for any timing related puposes..
-  startMillis = millis();
-
   // Start Serial
   Serial.begin(BAUDRATE);
-  Serial.println("Start");
+  delay(500);
+  ESP_LOGI(TAG,"Start setup");
+  WRITE_PERI_REG(RTC_CNTL_BROWN_OUT_REG, 0); // disable brownout detector
 
+  // for any timing related puposes..
+  state.startMillis = millis();
+  pins = new PINDEF();
+  ESP_LOGI(TAG,"getDefaultPins");
+  state.getDefaultPinDef(pins);
+  ESP_LOGI(TAG,"wifi.createJsonDoc()");
+  WifiController::createJsonDoc();
 
-  // if we boot for the first time => reset the preferences! // TODO: Smart? If not, we may have the problem that a wrong pin will block bootup
-  if (isFirstRun()) {
-    Serial.println("First Run, resetting config?");
-    resetConfigurations();
-  }
-
-  // check if setup went through after new config - avoid endless boot-loop
-  preferences.begin("setup", false);
-  if (preferences.getBool("setupComplete", true) == false) {
-    Serial.println("Setup not done, resetting config?"); //TODO not working!
-    resetConfigurations();
-  }
-  else {
-    Serial.println("Setup done, continue.");
-  }
-  preferences.putBool("setupComplete", false);
-  preferences.end();
-
-  // load config
-  loadConfiguration();
-
-  // display state
-  printInfo();
+  ESP_LOGI(TAG,"state.setup");
+  state.setup(pins, WifiController::getJDoc());
+  state.printInfo();
 
-  // reset jsonDocument
-  jsonDocument.clear();
+  ESP_LOGI(TAG,"Config::setup");
+  Config::setup(pins);
 
+  WifiController::getJDoc()->clear();
   // connect to wifi if necessary
-  bool isResetWifiSettings = false;
-  //autoconnectWifi(isResetWifiSettings);
-  initWifiAP(WifiSSIDAP);
-  setupRouting();
-  startServer();
-  init_Spiffs();
-
-  Serial.println(state_act_endpoint);
-  Serial.println(state_get_endpoint);
-  Serial.println(state_set_endpoint);
-
+  ESP_LOGI(TAG,"wifi.setup");
+  WifiController::setup();
 #ifdef IS_SLM
-  setup_slm();
+  ESP_LOGI(TAG,"IS_SLM");
+  slm.setup();
+#endif
+#ifdef IS_LED
+  ESP_LOGI(TAG,"IS_LED");
+#ifdef DEBUG_LED
+  led.DEBUG = true;
+#endif
+  led.setup(pins);
 #endif
 
-  setup_matrix();
-  setup_motor();
-  setup_laser();
+#ifdef IS_MOTOR
+  ESP_LOGI(TAG,"IS_MOTOR");
+#ifdef DEBUG_MOTOR
+  motor.DEBUG = true;
+#endif
+  motor.setup(pins);
+#endif
 
-  /*
-    setting up playstation controller
-  */
-  if(isFirstRun()) {
-    // 
-    clearBlueetoothDevice();
+#if defined IS_PS4 || defined IS_PS3
```

Comment:
Right now, removing the MAC addresses on first boot won'T work, right? Save to re-enable it?

Reply:
for me the clearBlueetoothDevice func crash. causing a bootloop

Reply:
Oh, that's not good :D 
I have changed it to do it only once: at first boot after flashing the new FW. However, this does not seem to solve the issue with multiple stored MAC addresses in the ESP in some cases..weird! Perhaps a PS4 controller issue?